lib/src/Uncertainty/Distribution/openturns/SquaredNormal.hxx

```diff
@@ -0,0 +1,109 @@
+/**
+ *  @brief The Squared Normal distribution
+ *
+ *  Copyright 2005-2019 Airbus-EDF-IMACS-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#ifndef OPENTURNS_SQUAREDNORMAL_HXX
+#define OPENTURNS_SQUAREDNORMAL_HXX
+
+#include "openturns/OTprivate.hxx"
+#include "openturns/ContinuousDistribution.hxx"
+#include "openturns/Normal.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+/**
+ * @class SquaredNormal
+ *
+ * The SquaredNormal (gaussian) distribution.
+ */
+class OT_API SquaredNormal
+  : public ContinuousDistribution
+{
+
+  CLASSNAME
+public:
+
+  /** Default constructor */
+  SquaredNormal();
+
+  /** Constructor */
+  SquaredNormal(Scalar mu, Scalar sigma);
+
+  /** Comparison operator */
+  Bool operator ==(const SquaredNormal & other) const;
+
+  /** String converter */
+  String __repr__() const;
+  String __str__(const String & offset = "") const;
+
+  /* Interface inherited from Distribution */
+
+  /** Virtual constructor */
+  virtual SquaredNormal * clone() const;
+
+  /** Get one realization of the SquaredNormal distribution */
+  Point getRealization() const;
+  Sample getSample(const UnsignedInteger size) const;
+
+  /** Get the CDF of the SquaredNormal distribution */
+  using ContinuousDistribution::computeCDF;
+  Scalar computeCDF(const Point & point) const;
+
+  /** Get the PDF of the distribution */
+  using ContinuousDistribution::computePDF;
+  Scalar computePDF(const Point & point) const;
+
+  /** Get the characteristic function of the distribution */
+  Complex computeCharacteristicFunction(const Scalar x) const;
+
+  /** Parameters value accessor */
+  virtual Point getParameter() const;
+  virtual void setParameter(const Point & parameters);
+
+  /** Parameters description accessor */
+  virtual Description getParameterDescription() const;
+
+  /** Method save() stores the object through the StorageManager */
+  void save(Advocate & adv) const;
+
+  /** Method load() reloads the object from the StorageManager */
+  void load(Advocate & adv);
+
+protected:
+
+private:
+
+  /** Compute the mean of the distribution */
+  void computeMean() const;
+
+  /** Compute the covariance of the distribution */
+  void computeCovariance() const;
+  
+  /** Compute the range */
+  void computeRange();
+  
+  Scalar mu_;
+  Scalar sigma_;
+  Normal normal_;
```

Comment:
You're duplicating data here, I'd prefer just mu, sigma

Reply:
Yes, it was an implementation choice in order to reuse Normal methods. I can recreate a Normal(mu_, sigma_) object where it is needed.

Reply:
Never do that! Creating a Normal object takes ages, it will ruin the performance of the distribution. You uses normal_ only to check equality (better check if mu_ == other.mu_ and sigma_ == other.sigma_) and for parameter management, which can be copy/paste from another distribution.
Sorry for being quite absent from the debate, I have urgent matters to solve before to go back to OT...

Reply:
done